test/form/index.js

```diff
@@ -0,0 +1,183 @@
+
+    instance.setValid(false);
+    assert.isFalse(instance.state.isValid, 'state was changed');
+
+    instance.addInput({ setPristine });
+    instance.setPristine(callback);
+
+    assert.deepEqual(instance.state, instance.getDefaultState(), 'state is back to default');
+    assert.isTrue(setPristine.calledOnce, 'setPristine on component was called');
+
+    setTimeout(() => {
+      assert.isTrue(callback.calledOnce, 'callback was called');
+    }, 100);
+    wrapper.unmount();
+  });
+
+  it('setValid', () => {
+    const props = { onValidityChange: fake() };
+    const wrapper = shallow(createElement(Form, props));
+    const instance = wrapper.instance();
+    const doneFake = fake();
+
+    assert.isTrue(instance.state.isValid, 'default state is valild');
+
+    instance.setValid(false, doneFake);
+
+    assert.isTrue(doneFake.calledOnce, 'callback was called');
+    assert.isTrue(props.onValidityChange.calledOnce, 'onValidityChange prop was called');
+    assert.isFalse(instance.state.isValid, 'state was changed');
+  });
+
+  it('isDisabled', () => {
+    // this.props.locked is true
+    const props = { locked: true };
+    const wrapperWithLock = shallow(createElement(Form, props));
+    const instanceWithLock = wrapperWithLock.instance();
+    assert.isTrue(instanceWithLock.isDisabled(), 'form is disabled if locked is true');
+
+    // this.state.isValid is false
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+
+    instance.setValid(false);
+    assert.isTrue(instance.isDisabled(), 'form is disabled if state is not valid');
+  });
+
+  it('updateValidity', () => {
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+    instance.setValid = fake();
+
+    instance.updateValidity();
+
+    assert.isTrue(instance.setValid.calledOnce, 'setValid was called');
+  });
+
+  it('reset', () => {
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+    const reset = fake();
+
+    instance.addInput({ reset });
+    instance.setValid(false);
+    assert.isFalse(instance.state.isValid, 'state was changed');
+
+    instance.reset();
+    assert.deepEqual(instance.state, instance.getDefaultState(), 'state is back to default');
+    assert.isTrue(reset.calledOnce, 'reset on component was called');
+  });
+
+  it('submit', () => {
+    const preventDefault = fake();
+    const wrapper = shallow(createElement(Form));
+
+    wrapper.find('form').simulate('submit', { preventDefault });
+    const onSubmit = wrapper.props().onSubmit;
+
+    assert.isTrue(preventDefault.calledOnce, 'preventDefault was called');
+    assert.exists(onSubmit, 'onSubmit prop exists');
+    assert.typeOf(onSubmit, 'function', 'onSubmit must be a funciton');
+  });
+
+  it('validate', () => {
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
```

Comment:
- create 2 instances
- add object with fake `validate` that returns resolved promise to 1 instance
- add another object with  fake `validate` that returns rejected promise to 2 instance
- test this 2 instances.